### PR TITLE
Convert `call_user_func` and `call_user_func_array` to variable functions, project wide

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- Convert `call_user_func` and `call_user_func_array` to variable functions, which are about four times faster and also easier to read.

### Testing Instructions

Code review only, should not change behavior

### Documentation Changes Required

None.

### CODE REVIEW

File: build/phpcs/Joomla/Sniffs/Commenting/FileCommentSniff.php [hunk 1]
```diff
@@ -459,7 +459,7 @@ protected function processTags($commentStart, $commentEnd)
             $method = 'process'.$tagName;
             if (method_exists($this, $method) === true) {
                 // Process each tag if a method is defined.
-                call_user_func(array($this, $method), $errorPos);
+	            $this->$method($errorPos);
```
Comment (andrepereiradasilva) [inline build/phpcs/Joomla/Sniffs/Commenting/FileCommentSniff.php]: not sure this should be changed since is a phpcs sniff

**Comment by andrepereiradasilva** [comment]
i add some doubts about this one, so applied the patch
Whent to Contet -> Categories
```
An error has occurred.
0 Call to a member function addSubmenu() on string
Call stack
#	Function	Location
1	JApplicationCms->execute()	/path/to/joomla-staging/administrator/index.php:51
2	JApplicationAdministrator->doExecute()	/path/to/joomla-staging/libraries/cms/application/cms.php:265
3	JApplicationAdministrator->dispatch()	/path/to/joomla-staging/libraries/cms/application/administrator.php:152
4	JComponentHelper::renderComponent()	/path/to/joomla-staging/libraries/cms/application/administrator.php:98
5	JComponentHelper::executeComponent()	/path/to/joomla-staging/libraries/cms/component/helper.php:376
6	require_once()	/path/to/joomla-staging/libraries/cms/component/helper.php:401
7	JControllerLegacy->execute()	/path/to/joomla-staging/administrator/components/com_categories/categories.php:27
8	CategoriesController->display()	/path/to/joomla-staging/libraries/legacy/controller/legacy.php:707
9	CategoriesHelper::addSubmenu()	/path/to/joomla-staging/administrator/components/com_categories/controller.php:94
```

**Comment by frankmayer** [comment]
Hmm, going to check, thanks.

**Comment by frankmayer** [comment]
@andrepereiradasilva the issue, you were seeing, shoudl be resolved now. Thanks!

**Comment by frankmayer** [comment]
Test (code review) please.  @Quy @andrepereiradasilva? Thanks

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on d59225f8b90b48524391fe642b26744945300970<br /><br />code review.
applied patch navigated a little in backend/frontend found no issues.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13257">issues.joomla.org/tracker/joomla-cms/13257</a>.</sub>

**Comment by frankmayer** [comment]
Thank you @andrepereiradasilva

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d59225f8b90b48524391fe642b26744945300970<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13257">issues.joomla.org/tracker/joomla-cms/13257</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**